chainer/training/triggers/minmax_value_trigger.py

```diff
@@ -0,0 +1,102 @@
+from chainer import reporter
+from chainer.training.trigger import IntervalTrigger
+
+
+class BestValueTrigger(object):
+
+    """Trigger invoked when specific value becomes best.
+
+    Args:
+        key (str): Key of value.
+        compare (function): Compare function which takes current best value and
+            new value and returns whether new value is better than current
+            best.
+        trigger: Trigger that decides the comparison interval between current
+            best value and new value. This must be a tuple in the form of
+            ``<int>, 'epoch'`` or ``<int>, 'iteration'`` which is passed to
+            :class`IntervalTrigger`.
+
+    """
+    def __init__(self, key, compare, trigger=(1, 'epoch')):
+        self._key = key
+        self._best_value = None
+        self._interval_trigger = IntervalTrigger(*trigger)
```

Comment:
You can use [get_trigger](https://github.com/pfnet/chainer/blob/master/chainer/training/trigger.py#L44) to create `Trigger` object. If you use this function, `trigger` argument can be either of a callable, None, or tuple.